presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -163,10 +163,15 @@ public void createTable(Table table)
     }
 
     @Override
-    public void dropTable(String databaseName, String name, boolean deleteData)
+    public void dropTableWithEnvironmentContext(String databaseName, String name, boolean deleteData, EnvironmentContext environmentContext)
             throws TException
     {
-        client.drop_table(databaseName, name, deleteData);
+        if (environmentContext.isSetProperties()) {
```

Comment:
can we avoid the `if` and always call `drop_table_with_environment_context` here?

(otherwise we should rethink the method name)